AssetInformationApi/V1/Gateways/DynamoDbGateway.cs

```diff
@@ -59,6 +61,11 @@ public async Task<Asset> GetAssetByAssetId(GetAssetByAssetIdRequest query)
         [LogCall]
         public async Task<Asset> AddAsset(AssetDb asset)
         {
+            if (PostcodeHelpers.IsValidPostCode(asset.AssetAddress.PostCode))
+            {
+                asset.AssetAddress.PostCode = PostcodeHelpers.NormalizePostcode(asset.AssetAddress.PostCode);
+            }
```

Comment:
What if the asset isnt valid? Do we not normalise it?

Reply:
Not sure how you mean, the added code just looks at whatever is the current postcode that comes within the request and normalizes it.

The way asset validation works is not changed by this PR. What changes would you make?

Reply:
In the gateway, you only normalise the postcode if its a valid postcode. But that scenario should never occur if the postcode is validated in the controller. I would remove the check, and just normalize everything